api_controller.py

```diff
@@ -180,28 +180,38 @@ def trigger_nickname(self, param_nickname, param_game_type=rating, days=7):
             self.save_play_info_to_sql(local_user_db, local_match_db, local_match_detail_db)
             print("DB Save End")
 
-    def continuous_collecting(self, rating_tier,
-                              ref_before_date: datetime.timedelta = datetime.timedelta(minutes=5),
-                              ref_later_date: datetime.timedelta = datetime.timedelta(minutes=1)):
+    def continuous_collecting(self, rating_tier, search_direction,
+                              ref_before_date: datetime.timedelta = datetime.timedelta(minutes=10)):
         player_id_list = self.__sql.select_by_rating(rating_tier)
+        if len(player_id_list) is 0:
+            return
+
+        search_date = self.__sql.select_search_date()
+        ref_date: datetime
+        other_date: datetime
+        if search_direction == "past":
+            before_date = search_date[rating_tier][0] - ref_before_date
+            later_date = search_date[rating_tier][0] + datetime.timedelta(minutes=1)
+            search_date[rating_tier][0] -= ref_before_date
+        elif search_direction == "recent":
+            if search_date[rating_tier][1] - ref_before_date < datetime.datetime.now():
+                before_date = search_date[rating_tier][1] - datetime.timedelta(minutes=1)
+                later_date = search_date[rating_tier][1] + ref_before_date
```

Comment:
I think it would be nice to pair variable names together.
For example, before_date : after_date or early_date : later_date..